aineko/core/runner.py

```diff
@@ -139,88 +139,65 @@ def prepare_datasets(
             }
 
-        # Fail if reserved dataset names are defined in catalog
         for reserved_dataset in DEFAULT_KAFKA_CONFIG.get("DATASETS"):
             if reserved_dataset in config:
                 raise ValueError(
                     f"Unable to create dataset `{reserved_dataset}`. "
                     "Reserved for internal use."
                 )
 
-        # Add logging dataset to catalog
-        config[
-            DEFAULT_KAFKA_CONFIG.get("LOGGING_DATASET")
-        ] = Config.Pipeline.Dataset(
-            type=AINEKO_CONFIG.get("KAFKA_STREAM_TYPE"),
-            params=DEFAULT_KAFKA_CONFIG.get("DATASET_PARAMS"),
-        )
-
-        # Create all dataset defined in the catalog
-        dataset_list = []
         for dataset_name, dataset_config in config.items():
             logger.info(
                 "Creating dataset: %s: %s", dataset_name, dataset_config
             )
-            # Create dataset for kafka streams
-            if dataset_config.type == AINEKO_CONFIG.get("KAFKA_STREAM_TYPE"):
-                # User defined
-                if not dataset_config.params:
-                    dataset_config.params = {}
-
-                dataset_params = {
-                    **DEFAULT_KAFKA_CONFIG.get("DATASET_PARAMS"),
-                    **dataset_config.params,
-                }
-
-                # Configure dataset
-                if self.dataset_prefix:
-                    topic_name = f"{self.dataset_prefix}.{dataset_name}"
-                else:
-                    topic_name = dataset_name
-
-                new_dataset = NewTopic(
-                    topic=topic_name,
-                    num_partitions=dataset_params.get("num_partitions"),
-                    replication_factor=dataset_params.get("replication_factor"),
-                    config=dataset_params.get("config"),
-                )
-
-                # Add dataset to appropriate list
-                dataset_list.append(new_dataset)
-
-            else:
-                raise ValueError(
-                    "Unknown dataset type. Expected: "
-                    f"{AINEKO_CONFIG.get('STREAM_TYPES')}."
-                )
-
-        # Create all configured datasets
-        datasets = kafka_client.create_topics(dataset_list)
+            # update dataset config here:
+            dataset: AbstractDataset = AbstractDataset.from_config(
+                dataset_name, dataset_config
+            )
+            datasets.append(dataset)
+
+        # Create logging dataset
+        logging_config = {
+            "location": "localhost:9092",
+            "type": "aineko.datasets.kafka.Kafka",
+        }
+        logging_dataset_name = DEFAULT_KAFKA_CONFIG.get("LOGGING_DATASET")
+        logger.info(
+            "Creating dataset: %s: %s", logging_dataset_name, logging_config
+        )
+        logging_dataset: AbstractDataset = AbstractDataset.from_config(
+            logging_dataset_name, logging_config
+        )
```

Comment:
The type hints don't seem correct. The datasets are not AbstractDataset in this context.

Reply:
This one is a bit tricky. They are all technically subclasses of AbstractDataset, which can by implied by the syntax. They aren't all necessarily a single type of subclass (like KafkaDataset) since we want the runner to load the most general case of datasets. I couldn't find a recommended way of handling this, other than referencing the generic superclass of AbstractDataset

Reply:
Thanks, I see how that makes sense.